src/pointer.rs

```diff
@@ -441,13 +475,29 @@ impl PartialEq<&str> for Pointer {
         &&self.0 == other
     }
 }
-
+impl<'p> PartialEq<String> for &'p Pointer {
+    fn eq(&self, other: &String) -> bool {
+        self.0.eq(other)
+    }
+}
 impl PartialEq<str> for Pointer {
     fn eq(&self, other: &str) -> bool {
         &self.0 == other
     }
 }
 
+impl PartialEq<Pointer> for &str {
+    fn eq(&self, other: &Pointer) -> bool {
+        *self == (&other.0)
```

Comment:
```suggestion
        *self == &other.0
```
or are the parentheses necessary?